Library/OptimizelyClientType.swift

```diff
@@ -1,9 +1,20 @@
 import Foundation
-import Optimizely
 
-public protocol OptimizelyClientType: class {
-  func activate(experimentKey: String, userId: String, attributes: OptimizelyAttributes?) throws -> String
+public protocol OptimizelyClientType: AnyObject {
+  func activate(experimentKey: String, userId: String, attributes: [String: Any?]?) throws -> String
 }
 
-extension OptimizelyClient: OptimizelyClientType {
+extension OptimizelyClientType {
+  public func variant(
```

Comment:
Just to avoid coalescing twice we can do:

```swift
public func variant(
  for experiment: OptimizelyExperiment.Key,
  userId: String
) -> OptimizelyExperiment.Variant {
  guard
    let variation = try? self.activate(experimentKey: experiment.rawValue, userId: userId, attributes: nil),
    let experiment = OptimizelyExperiment.Variant(rawValue: variation)
  else {
    return .control
  }

  return experiment
}
```